deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.kt

```diff
@@ -0,0 +1,580 @@
+            try {
+                urisTrie.addToTrie(uriTemplate, element.annotatedClass?.className?.reflectionName() ?: "", element.method)
+            } catch (e: IllegalArgumentException) {
+                logError(element = element.annotatedClass,
+                        message = e.message ?: "")
+            }
+            val deeplinkUri = DeepLinkUri.parseTemplate(uriTemplate)
+            // Keep track of pathVariables added in a module so that we can check at runtime to ensure
+            // that all pathVariables have a corresponding entry provided to BaseDeepLinkDelegate.
+            for (pathSegment: String in deeplinkUri.pathSegments()) {
+                if (isConfigurablePathSegment(pathSegment)) {
+                    pathVariableKeys.add(
+                        pathSegment.substring(
+                            configurablePathSegmentPrefix.length,
+                            pathSegment.length - configurablePathSegmentSuffix.length
+                        )
+                    )
+                }
+            }
+        }
+        val deepLinkRegistryBuilder = TypeSpec.classBuilder(
+            (className + REGISTRY_CLASS_SUFFIX)
+        ).addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+            .superclass(ClassName.get(BaseRegistry::class.java))
+        val stringMethodNames = getStringMethodNames(urisTrie, deepLinkRegistryBuilder)
+        val constructor = MethodSpec.constructorBuilder()
+            .addModifiers(Modifier.PUBLIC)
+            .addCode(
+                CodeBlock.builder()
+                    .add(
+                        "super(\$T.readMatchIndexFromStrings( new String[] {$stringMethodNames})",
+                        CLASS_UTILS
+                    ).build()
+            )
+            .addCode(generatePathVariableKeysBlock(pathVariableKeys))
+            .build()
+
+        // For debugging it is nice to have a file version of the index, just comment this in to get
+        // on in the classpath
+//    FileObject indexResource = filer.createResource(StandardLocation.CLASS_OUTPUT, "",
+//    MatchIndex.getMatchIdxFileName(className));
+//    urisTrie.writeToOutoutStream(indexResource.openOutputStream());
+        deepLinkRegistryBuilder.addMethod(constructor)
+        originatingElements.forEach { originatingElement ->
+            deepLinkRegistryBuilder.addOriginatingElement(originatingElement)
+        }
+        JavaFile.builder(packageName, deepLinkRegistryBuilder.build()).build()
+            .writeTo(environment.filer, XFiler.Mode.Aggregating)
+    }
+
+    private fun generatePathVariableKeysBlock(pathVariableKeys: Set<String>): CodeBlock {
+        val pathVariableKeysBuilder = CodeBlock.builder()
+        pathVariableKeysBuilder.add(
+            ",\n" + "new \$T[]{",
+            ClassName.get(String::class.java)
+        )
+        val pathVariableKeysArray = pathVariableKeys.toTypedArray()
+        for (i in pathVariableKeysArray.indices) {
```

Comment:
Looks like a couple places we need to construct a collection literal, I've created utilties like
```
fun constructCollectionLiteral(
    opening: CodeBlock,
    values: Collection<CodeBlock>,
    closing: CodeBlock,
    separator: CodeBlock = CodeBlock.of(",\n")
): CodeBlock {
    CodeBlock.builder().apply {
        add(opening)
        values.forEachIndexed { index, codeBlock ->
            add(codeBlock)
            if (index != values.size - 1) {
                add(separator)
            }
        }
        add(closing)
        return build()
    }
}
fun Collection<CodeBlock>.constructArrayLiteral() =
    constructCollectionLiteral(CodeBlock.of("[\n"), this, CodeBlock.of("\n]"))

```

Reply:
Let me look at that, the code generators are literally untouched from the conversion tool.

Reply:
Though this makes sense I'm going to leave it like this for this PR and mark that as a future improvement.